### DISCUSSION TITLE
Cannot deploy - Error! queryA EREFUSED api.zeit.co [WINDOWS]

### DISCUSSION THREAD

**Discussion by HelloKashif** [vercel/vercel#3756] [Help]
On Windows 10
Now CLI 17.0.1
Getting this error while calling now
Error! queryA EREFUSED api.zeit.co

**Comment by Alaboudi1** [comment]
I am also experiencing the same problem after updating to the new cli. I am using WSL 2 in windows. 

Now CLI 17.0.2
Error! queryA ETIMEOUT api.zeit.co

Update: downgrade now cli to version 16.x fixed the issue.

**Comment by styfle** [comment]
Are you still experiencing this issue? Does it happen every time?

**Comment by vuphan365** [comment]
Still happen :(

**Comment by lauroaraujo** [comment]
This is happening to me as well and downgrading did not help.

Mac OS X 10.14.6
Now CLI 17.0.4

```
$ now -A now.preview.json
Now CLI 17.0.4
Error! queryA EREFUSED api.zeit.co
```

The strange thing is that other commands work: `now secrets`, `now list`, etc

Even stranger is that **the error only happens at my work's network**, but there are no rules blocking zeit, and the **other commands (`now secrets`, `now list`) work normally on this network**...

I've tried logging out and in of zeit and deleting the .now folder, but neither helped

EDIT: Does `now deploy` use p2p by any chance? Any unconventional connection that could be blocked by a company's firewall?

**Comment by styfle** [comment]
No, nothing unconventional. Perhaps the issue is related to Anycast:

> Each request to any `api.zeit.co` endpoint, will be automatically resolved to the closest Origin location using Anycast. 

https://zeit.co/docs/api

**Comment by williamluke4** [comment]
Same issue using Now CLI `17.0.4` on Windows

EDIT: Similar to @lauroaraujo I change my network and it deployed.. Very weird

**Comment by styfle** [comment]
If you add the `--debug` flag the you can see each network request.

Maybe that will provide more insight.

**Comment by williamluke4** [comment]
@styfle Here is the output when using `now --debug`
```
C:\Users\willi\Github\PrismaFlow>now --debug
> [debug] [2020-03-12T15:55:22.906Z] Found config in file C:\Users\willi\Github\PrismaFlow\now.json
Now CLI 17.0.4
> [debug] [2020-03-12T15:55:22.933Z] user supplied no target, defaulting to deploy
> [debug] [2020-03-12T15:55:22.945Z] Spinner invoked (Retrieving scope…) with a 1000ms delay
> [debug] [2020-03-12T15:55:24.366Z] Spinner ended (Retrieving scope…)
> [debug] [2020-03-12T15:55:25.380Z] GET https://api.zeit.co/teams 
> [debug] [2020-03-12T15:55:26.375Z] GET https://api.zeit.co/teams : 993.268ms
(node:6736) Warning: No such label '> [debug] #1 GET /teams' for console.timeEnd()
> [debug] [2020-03-12T15:55:26.427Z] GET https://api.zeit.co/teams 
(node:6736) Warning: Label '> [debug] #1 GET /teams}' already exists for console.time()
> [debug] [2020-03-12T15:55:27.451Z] GET https://api.zeit.co/teams : 988.034ms
(node:6736) Warning: No such label '> [debug] #1 GET /teams' for console.timeEnd()
> [debug] [2020-03-12T15:55:27.669Z] Spinner invoked (Retrieving project…) with a 1000ms delay
> [debug] [2020-03-12T15:55:27.673Z] GET https://api.zeit.co/teams 
> [debug] [2020-03-12T15:55:27.694Z] GET https://api.zeit.co/projects/QmRV1Wj8wHtRJYoLTfSXXbViVybbUxeM48bJQFrihhZoyw?teamId=team_pzzTUy8TWCB9zVj2gpx2JyAv 
(node:6736) Warning: Label '> [debug] #1 GET /teams}' already exists for console.time()
> [debug] [2020-03-12T15:55:28.680Z] GET https://api.zeit.co/teams : 988.767ms
(node:6736) Warning: No such label '> [debug] #1 GET /teams' for console.timeEnd()
> [debug] [2020-03-12T15:55:28.712Z] GET https://api.zeit.co/projects/QmRV1Wj8wHtRJYoLTfSXXbViVybbUxeM48bJQFrihhZoyw?teamId=team_pzzTUy8TWCB9zVj2gpx2JyAv : 1016.165ms
> [debug] [2020-03-12T15:55:28.728Z] Spinner ended (Retrieving project…)
> [debug] [2020-03-12T15:55:28.734Z] Spinner invoked (Deploying atto-byte/prisma-flow) with a 0ms delay
[now-client-debug] 2020-03-12T15:55:28.740Z Creating deployment...
[now-client-debug] 2020-03-12T15:55:28.783Z Provided 'path' is a directory. Reading subpaths... 
[now-client-debug] 2020-03-12T15:55:28.786Z Read 22 subpaths
[now-client-debug] 2020-03-12T15:55:28.798Z Found 1 rules in .nowignore
[now-client-debug] 2020-03-12T15:55:28.828Z Building file tree...
[now-client-debug] 2020-03-12T15:55:28.919Z Read 38 files in the specified directory
[now-client-debug] 2020-03-12T15:55:29.120Z Yielding a 'hashes-calculated' event with 38 hashes
[now-client-debug] 2020-03-12T15:55:29.122Z Using provided API URL: https://api.zeit.co
[now-client-debug] 2020-03-12T15:55:29.123Z Using provided user agent: now 17.0.4 node-v12.0.0 win32 (x64)
[now-client-debug] 2020-03-12T15:55:29.124Z Setting platform version to 2
[now-client-debug] 2020-03-12T15:55:29.147Z Creating the deployment and starting upload...
[now-client-debug] 2020-03-12T15:55:29.149Z Determining necessary files for upload...
[now-client-debug] 2020-03-12T15:55:29.151Z Creating deployment
[now-client-debug] 2020-03-12T15:55:29.154Z Sending deployment creation API request
[now-client-debug] 2020-03-12T15:55:57.784Z Yielding a 'error' event
> [debug] [2020-03-12T15:55:57.787Z] Spinner ended (Deploying atto-byte/prisma-flow)
> [debug] [2020-03-12T15:55:58.331Z] Error: Error: queryA EREFUSED api.zeit.co
Error: queryA EREFUSED api.zeit.co
    at Now.handleDeploymentError (C:\Users\willi\AppData\Roaming\nvm\v12.0.0\node_modules\now\dist\index.js:2:1825241)
    at processDeployment (C:\Users\willi\AppData\Roaming\nvm\v12.0.0\node_modules\now\dist\index.js:2:2279034)
    at Now.create (C:\Users\willi\AppData\Roaming\nvm\v12.0.0\node_modules\now\dist\index.js:2:1822725)
    at createDeploy (C:\Users\willi\AppData\Roaming\nvm\v12.0.0\node_modules\now\dist\index.js:2:2012932)
    at Module.main (C:\Users\willi\AppData\Roaming\nvm\v12.0.0\node_modules\now\dist\index.js:2:2862946)
    at Se (C:\Users\willi\AppData\Roaming\nvm\v12.0.0\node_modules\now\dist\index.js:2:1728252)
Error! queryA EREFUSED api.zeit.co

```

**Comment by hynra** [comment]
Same issue using Now CLI 19.0.1 on Windows
```
[now-client-debug] 2020-05-18T07:44:52.676Z Creating deployment...
[now-client-debug] 2020-05-18T07:44:52.677Z Provided 'path' is a directory. Reading subpaths...
[now-client-debug] 2020-05-18T07:44:52.678Z Read 4 subpaths
[now-client-debug] 2020-05-18T07:44:52.681Z Found 1 rules in .nowignore
[now-client-debug] 2020-05-18T07:44:52.681Z Building file tree...
[now-client-debug] 2020-05-18T07:44:52.691Z Read 3 files in the specified directory
[now-client-debug] 2020-05-18T07:44:52.693Z Yielding a 'hashes-calculated' event with 3 hashes
[now-client-debug] 2020-05-18T07:44:52.694Z Using provided API URL: https://api.vercel.com
[now-client-debug] 2020-05-18T07:44:52.694Z Using provided user agent: now 19.0.1 node-v10.17.0 win32 (x64)
[now-client-debug] 2020-05-18T07:44:52.695Z Setting platform version to 2
[now-client-debug] 2020-05-18T07:44:52.695Z Creating the deployment and starting upload...
[now-client-debug] 2020-05-18T07:44:52.696Z Determining necessary files for upload...
[now-client-debug] 2020-05-18T07:44:52.697Z Creating deployment
[now-client-debug] 2020-05-18T07:44:52.698Z Sending deployment creation API request
[now-client-debug] 2020-05-18T07:45:11.698Z Yielding a 'error' event
> [debug] [2020-05-18T07:45:11.699Z] Spinner ended (Setting up project)
> [debug] [2020-05-18T07:45:12.332Z] Error: Error: queryA EREFUSED api.vercel.com
Error: queryA EREFUSED api.vercel.com
    at Now.handleDeploymentError (C:\Users\iman\AppData\Roaming\npm\node_modules\vercel\dist\index.js:2:1756758)
    at processDeployment (C:\Users\iman\AppData\Roaming\npm\node_modules\vercel\dist\index.js:2:2184846)
Error! queryA EREFUSED api.vercel.com
PS C:\Users\my\Documents\vercel-integration\DEPLOYMENT\chat-prod>
```

**Comment by ghost** [comment]
Same issue using
- node v12.20.1
- vercel 21.0.1

```
Vercel CLI 21.0.1
> [debug] [2021-01-12T03:49:22.612Z] user supplied no target, defaulting to deploy
> [debug] [2021-01-12T03:49:22.681Z] GET https://api.vercel.com/www/user 
> [debug] [2021-01-12T03:49:23.364Z] GET https://api.vercel.com/www/user : 683.025ms
> [debug] [2021-01-12T03:49:23.390Z] Spinner invoked (Retrieving project…) with a 1000ms delay
> [debug] [2021-01-12T03:49:23.392Z] GET https://api.vercel.com/projects/QmQmKdky3xnX3SJUhYHHtFte3QGMvsLFLgrrfAZpu1hibV? 
> [debug] [2021-01-12T03:49:23.849Z] GET https://api.vercel.com/projects/QmQmKdky3xnX3SJUhYHHtFte3QGMvsLFLgrrfAZpu1hibV? : 456.408ms
> [debug] [2021-01-12T03:49:23.854Z] Spinner ended (Retrieving project…)
> [debug] [2021-01-12T03:49:23.857Z] Spinner invoked (Deploying utadachiren/talent-db-web-talent) with a 0ms delay
[now-client-debug] 2021-01-12T03:49:23.858Z Creating deployment...
[now-client-debug] 2021-01-12T03:49:23.858Z Provided 'path' is a directory. Reading subpaths...
[now-client-debug] 2021-01-12T03:49:23.859Z Read 21 subpaths
[now-client-debug] 2021-01-12T03:49:23.862Z Found 25 rules in .vercelignore
[now-client-debug] 2021-01-12T03:49:23.862Z Building file tree...
[now-client-debug] 2021-01-12T03:49:23.871Z Found 57 files in the specified directory
[now-client-debug] 2021-01-12T03:49:23.891Z Yielding a 'hashes-calculated' event with 57 hashes
[now-client-debug] 2021-01-12T03:49:23.894Z Using provided API URL: https://api.vercel.com
[now-client-debug] 2021-01-12T03:49:23.894Z Using provided user agent: vercel 21.0.1 node-v12.20.1 win32 (x64)
[now-client-debug] 2021-01-12T03:49:23.895Z Setting platform version to harcoded value 2
[now-client-debug] 2021-01-12T03:49:23.895Z Creating the deployment and starting upload...
[now-client-debug] 2021-01-12T03:49:23.895Z Determining necessary files for upload...
[now-client-debug] 2021-01-12T03:49:23.896Z Creating deployment
[now-client-debug] 2021-01-12T03:49:23.898Z Sending deployment creation API request
[now-client-debug] 2021-01-12T03:49:54.880Z Yielding a 'error' event
> [debug] [2021-01-12T03:49:54.880Z] Spinner ended (Deploying utadachiren/talent-db-web-talent)
> [debug] [2021-01-12T03:49:54.945Z] Error: Error: queryA EREFUSED api.vercel.com
Error: queryA EREFUSED api.vercel.com
    at Now.handleDeploymentError (C:\Program Files (x86)\Nodist\bin\node_modules\vercel\dist\index.js:209718:12)
    at processDeployment (C:\Program Files (x86)\Nodist\bin\node_modules\vercel\dist\index.js:221796:41)
    at Now.create (C:\Program Files (x86)\Nodist\bin\node_modules\vercel\dist\index.js:209580:24)
    at createDeploy (C:\Program Files (x86)\Nodist\bin\node_modules\vercel\dist\index.js:209239:12)
    at main (C:\Program Files (x86)\Nodist\bin\node_modules\vercel\dist\index.js:203390:18)
    at main (C:\Program Files (x86)\Nodist\bin\node_modules\vercel\dist\index.js:209082:16)
Error! queryA EREFUSED api.vercel.com
```

**Comment by mrChernicharo** [comment]
same issue here

❯ vercel --prod --debug
Vercel CLI 21.3.3
> [debug] [2021-04-08T05:11:13.946Z] user supplied a possible target for deployment
> [debug] [2021-04-08T05:11:14.148Z] GET https://api.vercel.com/www/user 
> [debug] [2021-04-08T05:11:14.818Z] GET https://api.vercel.com/www/user : 667.479ms
> [debug] [2021-04-08T05:11:14.907Z] Spinner stopped (Retrieving project…)
> [debug] [2021-04-08T05:11:14.907Z] Spinner invoked (Retrieving project…) with a 1000ms delay
> [debug] [2021-04-08T05:11:14.912Z] GET https://api.vercel.com/projects/prj_P2CTFrHD7uzpNWKQ3ve4I5rrgKkq? 
> [debug] [2021-04-08T05:11:16.123Z] GET https://api.vercel.com/projects/prj_P2CTFrHD7uzpNWKQ3ve4I5rrgKkq? : 1210.950ms
> [debug] [2021-04-08T05:11:16.129Z] Setting target to production
> [debug] [2021-04-08T05:11:16.132Z] Spinner stopped (Deploying mrchernicharo/next-portfolio)
> [debug] [2021-04-08T05:11:16.132Z] Spinner invoked (Deploying mrchernicharo/next-portfolio) with a 0ms delay
[client-debug] 2021-04-08T05:11:16.133Z Creating deployment...
[client-debug] 2021-04-08T05:11:16.133Z Provided 'path' is a directory. Reading subpaths... 
[client-debug] 2021-04-08T05:11:16.135Z Read 16 subpaths
[client-debug] 2021-04-08T05:11:16.143Z Found 25 rules in .vercelignore
[client-debug] 2021-04-08T05:11:16.143Z Building file tree...
[client-debug] 2021-04-08T05:11:16.174Z Found 150 files in the specified directory
[client-debug] 2021-04-08T05:11:16.376Z Yielding a 'hashes-calculated' event with 150 hashes
[client-debug] 2021-04-08T05:11:16.377Z Using provided API URL: https://api.vercel.com
[client-debug] 2021-04-08T05:11:16.378Z Using provided user agent: vercel 21.3.3 node-v12.20.0 darwin (x64)
[client-debug] 2021-04-08T05:11:16.378Z Setting platform version to harcoded value 2
[client-debug] 2021-04-08T05:11:16.378Z Creating the deployment and starting upload...
[client-debug] 2021-04-08T05:11:16.379Z Determining necessary files for upload...
[client-debug] 2021-04-08T05:11:16.381Z Creating deployment
[client-debug] 2021-04-08T05:11:16.385Z Sending deployment creation API request
[client-debug] 2021-04-08T05:11:38.857Z Yielding a 'error' event
> [debug] [2021-04-08T05:11:38.988Z] Error: Error: queryA ESERVFAIL api.vercel.com
Error: queryA ESERVFAIL api.vercel.com
    at Now.handleDeploymentError (/Users/felipechernicharo/.nvm/versions/node/v12.20.0/lib/node_modules/vercel/dist/index.js:214920:12)
    at processDeployment (/Users/felipechernicharo/.nvm/versions/node/v12.20.0/lib/node_modules/vercel/dist/index.js:228500:41)
    at Now.create (/Users/felipechernicharo/.nvm/versions/node/v12.20.0/lib/node_modules/vercel/dist/index.js:214782:24)
    at createDeploy (/Users/felipechernicharo/.nvm/versions/node/v12.20.0/lib/node_modules/vercel/dist/index.js:214437:12)
    at main (/Users/felipechernicharo/.nvm/versions/node/v12.20.0/lib/node_modules/vercel/dist/index.js:208623:18)
    at main (/Users/felipechernicharo/.nvm/versions/node/v12.20.0/lib/node_modules/vercel/dist/index.js:214280:16)
Error! queryA ESERVFAIL api.vercel.com

**Comment by mrChernicharo** [comment]
Solved the problem setting the correct Node version on Vercel's dashboard

![Screen Shot 2021-04-08 at 02 18 44](https://user-images.githubusercontent.com/52944886/113972420-fa3afa80-9810-11eb-9b5c-9539960a6ea5.png)

Version 14 was the default while my project uses Node version 12.

Hope it helps

**Comment by axelson** [comment]
Okay, I figured this out. I ran into this error because my primary DNS server is non-recursive. It seems that whatever HTTP client vercel is using doesn't support non-recursive DNS servers.

**Reply by mbasaran** [reply]
Same error I was getting in docker (GitLab CI) Error: queryA EREFUSED [api.vercel.com](http://api.vercel.com/)

I added authoritative DNS of vercel to the container, it solved the problem:
 - echo "nameserver 198.51.44.13" > /etc/resolv.conf 

deploy_production:
  stage: deploy
  only:
    - main
  before_script:
    - export VERCEL_ORG_ID=$VERCEL_ORG_ID VERCEL_PROJECT_ID=$VERCEL_PROJECT_ID_PROD
  script:
    - npm install --global vercel
    - vercel pull --yes --environment=production --token=$VERCEL_TOKEN
    - vercel build --prod --token=$VERCEL_TOKEN
    - echo "nameserver 198.51.44.13" > /etc/resolv.conf #to fix vercel non-recursive DNS bug
    - vercel deploy --prebuilt --prod --token=$VERCEL_TOKEN
